src/Microsoft.Diagnostics.DebugServices/ServiceExportAttribute.cs

```diff
@@ -9,6 +9,7 @@ namespace Microsoft.Diagnostics.DebugServices
     public enum ServiceScope
     {
         Global,
+        Provider,
```

Comment:
This is the only part that is still not fully clear to me - as in how it lines up with the other ones. Essentially the others tell you at what level of the stack you want to make a service available. This one feels like you can request them at any point by type, so not sure if it's all that different than say global ones except with multiple registration. I wonder if it's better to introduce a separate attribute that instead of `ServiceExport` (since these are not behaving like classic services) is something like ServiceData and you can tell to what interfaces it gets injected to (like ServiceImport does), and then inject them as needed and requested.

Reply:
To change to a new attribute will require some work I'd rather not do right now.  A new function would have to be added to the service manager and the attr parsing code changed.

Reply:
Maybe we should keep track of cleanups needed in some issue, and can you please document the provider scope in the MD

Reply:
"MD"?

Reply:
command unalias b
command unalias r
command unalias t
command unalias p
command unalias print
command alias bp breakpoint set
command alias bl breakpoint list
command alias bc breakpoint delete
command alias bd breakpoint disable
command alias be breakpoint enable
command alias r register read
command alias ls _regexp-list
command alias k thread backtrace
command alias t thread step-in
command alias dv frame variable -f x -D 0
command alias gu thread step-out
command alias p thread step-over
command alias g process continue
command alias u disassemble -b
command alias ln target modules lookup
command alias db memory read -c 64
command alias dw memory read -c 32 -f x
command alias dd memory read -c 32 -f x
command alias dq memory read -c 16 -f p
command alias dp memory read -c 16 -f p
command alias lm target modules list -h -f
command alias print expr -f x -P 1 --
command alias threads thread list
command alias modules target modules list -f
command alias sig process handle -s false SIGUSR1 SIGUSR2 SIGABRT SIGINT SIGSEGV SIGBUS SIGILL SIGFPE
command alias lldbproc platform process list -s lldb
command alias corerunproc platform process list -n corerun

settings set interpreter.prompt-on-quit false
settings set target.x86-disassembly-flavor intel

settings set thread-format thread #${thread.index}: tid = ${thread.id}(${thread.id%tid}){ pc = ${frame.pc} sp = ${frame.sp}}{ ${module.file.basename}{`${function.name-with-args}${function.pc-offset}}}{ at ${line.file.basename}:${line.number}}{, name = '${thread.name}'}{, queue = '${thread.queue}'}{, activity = '${thread.info.activity.name}'}{, ${thread.info.trace_messages} messages}{, stop reason = ${thread.stop-reason}}{\nReturn value: ${thread.return-value}}{\nCompleted expression: ${thread.completed-expression}}\n

settings set disassembly-format "${current-pc-arrow}${addr-file-or-load}\t"

settings set frame-format "frame #${frame.index}: ${frame.sp} ${frame.fp} ${frame.pc}{ ${module.file.basename}{`${function.name-with-args}${function.pc-offset}}}{ at ${line.file.basename}:${line.number}}\n"

Reply:
You meant the extension doc, I assume.  I may want to change this in the future so it may be a waste to doc it right now.